### PR TITLE
Pade2 bug fix

### PR DESCRIPTION
**Pull request by shivupa**
## Proposed changes
The manual describes a pade2 functor (at least for 1 body Jastrows). At some point along the way this functionality got dropped. This puts it back in and cleans some things up.
This builds upon #3243 so this doesn't need to be reviewed until that is sorted out.

## What type(s) of changes does this code introduce?

- Code style update (formatting, renaming)
- Testing changes (e.g. new unit/integration/performance tests)

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?
Ryzen workstation
## Checklist
- Yes. This PR is up to date with current the current state of 'develop'
- Yes. Code added or changed in the PR has been clang-formatted
- Yes. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- No. Documentation has been added (if appropriate)

### CODE REVIEW

**Comment by shivupa** [comment]
I see this breaks complex builds. I'll take another look

File: src/Numerics/OptimizableFunctorBase.h [hunk 1]
```diff
@@ -117,7 +117,7 @@ struct OptimizableFunctorBase
   virtual void setGridManager(bool willmanage) {}
 };
 
-void print(OptimizableFunctorBase& func, std::ostream& os);
+void print(OptimizableFunctorBase& func, std::ostream& os, double extent = -1.0);
```
Comment (ye-luo) [inline src/Numerics/OptimizableFunctorBase.h:120]: Could you document this function, especially `extent`?

File: src/QMCWaveFunctions/Jastrow/PadeFunctors.h [hunk 2]
```diff
@@ -243,11 +243,13 @@ struct PadeFunctor : public OptimizableFunctorBase
       if (ia > -1)
       {
         int i = 0;
-        if (Opt_A) {
-          A = std::real( myVars[i++] = active[ia++] );
+        if (Opt_A)
+        {
+          A = std::real(myVars[i++] = active[ia++]);
```
Comment (ye-luo) [inline src/QMCWaveFunctions/Jastrow/PadeFunctors.h:248]: Remove {} around a single line

File: src/QMCWaveFunctions/Jastrow/PadeFunctors.h [hunk 3]
```diff
@@ -243,11 +243,13 @@ struct PadeFunctor : public OptimizableFunctorBase
...
         }
-        if (Opt_B) {
-          B0 = std::real( myVars[i] = active[ia] );
+        if (Opt_B)
+        {
```
Comment (ye-luo) [inline src/QMCWaveFunctions/Jastrow/PadeFunctors.h:251]: Remove {} around a single line

**Review by ye-luo — COMMENTED** [review]
Minor change and documentation request

File: src/QMCWaveFunctions/Jastrow/PadeFunctors.h [hunk 4]
```diff
@@ -287,25 +285,27 @@ struct Pade2ndOrderFunctor : public OptimizableFunctorBase
...
 
   /**@param r the distance
-    @return \f$ u(r) = a*r/(1+b*r) \f$
+    @return \f$ u(r) = \frac{a*r+c*r^2}{1+b*r} \f$
     */
-  inline real_type evaluate(real_type r)
+  inline real_type evaluate(real_type r) const
   {
-    real_type br(B * r);
-    return (A + br) * r / (1.0 + br);
```
Comment (ye-luo) [inline src/QMCWaveFunctions/Jastrow/PadeFunctors.h:299]: surprising bug got fixed

**Review by ye-luo — APPROVED** [review]

File: src/QMCWaveFunctions/Jastrow/PadeFunctors.h (hunk 2 above)
Comment (shivupa) [reply src/QMCWaveFunctions/Jastrow/PadeFunctors.h:247]: Removed. Sorry I always do this...

**Comment by ye-luo** [comment]
Test this please

**Pull request merged**